dclab/rtdc_dataset/plugins/plugin_feature.py

```diff
@@ -1,27 +1,65 @@
+"""
+.. versionadded:: 0.34.0
+"""
 
-# Create PlugInFeature class which is child of AncillaryFeature
-# it should have methods that can be overridden by a script
-# The script should create a child class of PlugInFeature
-# The methods defined in the script class will override the
-# PlugInFeature class.
-
-import abc
+import pathlib
+import importlib
+import sys
 
 from ..ancillaries import AncillaryFeature
 
 
-class PlugInFeature(AncillaryFeature, abc.ABC):
-    def __init__(self, *args, **kwargs):
-        super(PlugInFeature, self).__init__(*args, **kwargs)
+_registered_plugin_instances = []
+
+
+def create_new_plugin_feature(plugin_path):
+    """find an instanciate a PlugInFeature from a user-defined script"""
+    info = find_plugin_feature_script(plugin_path)
+    plugin_list = []
+    for ii in range(len(info["feature names"])):
+        ref_info = {
+            "feature_name": info["feature names"][ii],
+            "method": info["method"],
+            "req_config": info["config required"][ii],
+            "req_features": info["features required"][ii],
+            "priority": info["priority"],
+        }
+        plugin_list.append(PlugInFeature(**ref_info))
+        # add feature label etc
+    return plugin_list
+
+
+def find_plugin_feature_script(plugin_path):
+    # find script, return info dict
+    path = pathlib.Path(plugin_path)
+    # insert the plugin directory to sys.path so we can import it
+    sys.path.insert(-1, str(path.parent))
+    plugin = importlib.import_module(path.stem)
+    # undo our path insertion
+    sys.path.pop(0)
+    return plugin.info
+
+
+def remove_plugin_feature(plugin_instance):
+    """Convenience function for removing a plugin instance"""
+    PlugInFeature.features.remove(plugin_instance)
```

Comment:
Please add a check that `plugin_instance` is an instance of `PluginFeature`. And in a test, make sure that removing an `AncillaryFeature` does not work.